optimum/neuron/modeling_base.py

```diff
@@ -0,0 +1,440 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""NeuronModel base classe for inference on neuron devices using the same API as Transformers."""
+
+import logging
+import shutil
+from contextlib import contextmanager
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import TYPE_CHECKING, List, Optional, Union
+
+import torch
+from huggingface_hub import HfApi, HfFolder, hf_hub_download
+from transformers import AutoConfig, AutoModel
+
+from ..exporters.neuron import export
+from ..exporters.neuron.model_configs import *  # noqa: F403
+from ..exporters.tasks import TasksManager
+from ..modeling_base import OptimizedModel
+from ..utils.save_utils import maybe_load_preprocessors, maybe_save_preprocessors
+from .utils import NEURON_FILE_NAME, is_neuron_available, store_compilation_config
+
+
+if TYPE_CHECKING:
+    from transformers import PretrainedConfig
+
+    from ..exporters.neuron import NeuronConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+class NeuronModel(OptimizedModel):
```

Comment:
Maybe we should call this `NeuronModelBase` as it does not implement a `forward` method. Just `NeuronModel` would make me think it's the equivalent of `AutoModel`.

Reply:
Sounds good, will change!